tests/assertions/is_substring_of.zsh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env zunit
+
```

Comment:
Yeah it does. Admittedly, that's a bit of a hack to get the `is_empty` and `is_not_empty` rules to work, but I couldn't find any better way of doing it because the function doesn't recognise an empty string as an argument.

Looking at the docs [here](http://zsh.sourceforge.net/Doc/Release/Conditional-Expressions.html#Conditional-Expressions), it reads as `string = pattern`, in which case it makes sense that the wildcard only works on the right, and following that `assert "" is_substring 'some string'` *should* pass.

In reality though, if we did make the function pass for an empty value, that would let a lot of false positives fall through the tests when a variable was not defined, so I'm inclined to leave it as it is. Do you agree?